portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,11 +56,23 @@ private String arrayToJSON(List<Comment> strList) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    if (!dataLoaded) {
-        loadCommentData();
-        dataLoaded = true;
+    if (!loaded) {
+        loadCommentData(request);
     }
-    String json = arrayToJSON(comments);
+    int requestAmount;
+    try {
+        requestAmount = Integer.parseInt(request.getParameter("amount"));
+    } catch (NumberFormatException e) {
+        requestAmount = 0;
+    }
+    if(requestAmount < 0) {
+        requestAmount = 0;
+    }
```

Comment:
If there's any invalid requestAmount, the default page size should be used, instead of raising failures to users?

Reply:
Pushed a quick fix and now have a default value that can be changed if necessary